docs/admin.rst

```diff
@@ -1,6 +1,71 @@
 Administrating Treeherder
 =========================
 
+Changing the database models / tables
+-------------------------------------
+
+Most table-level modifications can be achieved by simply modifying
+the appropriate Treeherder model classes and generating Django
+schema migrations. However, there are some MySQL-specific gotchas
+which apply when performing these operations on larger tables (i.e.
+those with millions of rows or more).
+
+* Django's schema migrations are often rather inefficient for the
+  case where you are modifying multiple columns on a table at the
+  same time (it will generate as many alter table statements as
+  there are changes, when they could all be consolidated into one)
+* Altering a column with a foreign key relationship (i.e. changing
+  its type or adding/removing a null constraint) will cause MySQL
+  to re-check the validity of that relation for every row, which
+  can be quite expensive.
+* `It does not seem to be possible <https://dev.mysql.com/doc/refman/5.6/en/innodb-create-index-limitations.html>`_ to add or alter a table column
+  while additional rows are being added to the table without
+  creating duplicates (which will then cause the whole operation
+  to fail due to integrity errors)
+
+For this reason, it is recommended that you bypass using Django
+migrations and follow this procedure when modifying any large table:
+
+* Generate the SQL in a vagrant instance for the schema changes
+  you want to make (`./manage.py sqlmigrate model <name of migration>`
+  will print this out to the console) and modify if needed (i.e.
+  consolidating `ALTER TABLE` statements). For auditing purposes,
+  please post this SQL to an attachment on the bug (it may also
+  be a good idea to have another Treeherder dev look it over).
```

Comment:
I think that editing the migration file to use `RunSQL` with the custom SQL should be preferred unless there is some strong reason not to take this approach. This can be reviewed as a normal part of the patch. It can also do the `SET FOREIGN_KEY_CHECKS=0` part.